backend/routes/league.go

```diff
@@ -1,11 +1,94 @@
 package routes
 
-// s := r.PathPrefix("/league").Subrouter()
-// // "/products/"
-// s.HandleFunc("/postleague", AddLeague)
-// // "/products/{key}/"
-// s.HandleFunc("/getLeagueById/{ID}", GetLeagueById)
-// // "/products/{key}/details"
-// s.HandleFunc("/deleteLeagueById/{ID}", DeleteLeague)
-// // "/league/deleteLeagueById/{ID}"
-// s.HandleFunc("/updateLeagueById/{ID}", UpdateLeague)
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/grommers00/civitas/backend/internal"
+	"github.com/grommers00/civitas/backend/models"
+
+	"github.com/gorilla/mux"
+)
+
+// ConnectLeaguesSubrouter adds the different restFUL apis to the main router
+func ConnectLeaguesSubrouter(r *mux.Router) *mux.Router {
+	// "/leagues/"
+	f := r.PathPrefix("/leagues").Subrouter()
+
+	// "/leagues/"
+	f.HandleFunc("/", AddLeagues).Methods("POST")
+
+	// "/leagues/"
+	f.HandleFunc("/", GetAllLeagues).Methods("GET")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", GetLeaguesByID).Methods("GET")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", DeleteLeaguesByID).Methods("DELETE")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", UpdateLeaguesByID).Methods("PATCH")
+
+	return f
```

Comment:
You'll see in the other Connect-router functions that I removed *mux.Router return statement. That was after learning that we are passing it in correctly and that `r` inherits `f` subrouter. Let's reference the others and make the change here.

Reply:
fixed